source/common/http/conn_pool_grid.cc

```diff
@@ -27,11 +27,16 @@ absl::string_view describePool(const ConnectionPool::Instance& pool) {
 static constexpr uint32_t kDefaultTimeoutMs = 300;
 
 std::string getSni(const Network::TransportSocketOptionsConstSharedPtr& options,
-                   Network::UpstreamTransportSocketFactory& transport_socket_factory) {
+                   Upstream::HostConstSharedPtr& host) {
   if (options && options->serverNameOverride().has_value()) {
     return options->serverNameOverride().value();
   }
-  return std::string(transport_socket_factory.defaultServerNameIndication());
+  std::string transport = std::string(host->transportSocketFactory().defaultServerNameIndication());
+  if (!transport.empty() ||
+      !Runtime::runtimeFeatureEnabled("envoy.reloadable_features.allow_alt_svc_for_ips")) {
```

Comment:
IPs are illegal in the TLS SNI extension so if this method were actually returning the SNI, this would be bad. But it seems like that's not really what this method returns. Instead it's returning something like the target hostname? So how 'bout we rename this method to `getTargetHostname()`? Or something better?

Reply:
SGTM